### PR TITLE
Nano: Update Nano PyTorch quick start doc

### PR DESCRIPTION
**Pull request by y199387**

### CODE REVIEW

**Comment by yangw1234** [comment]
@zhentaocc could you help review the inference tutorial?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+])
+train_data = datasets.MNIST(
+    root=".",
+    train=True,
+    transform=data_transform
+)
+data_loader = DataLoader(train_data)
+# (Optional) Something else, like training ...
+trainer = Trainer()
+trainer.fit(model, data_loader)
```
Comment (zhentaocc) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md:50]: Would it be better to load a pre-trained model instead of training from scratch here.

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_openvino.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+When you're ready, you can simply append the following part to enable your OpenVINO acceleration.
+```python
+# trace your model as an OpenVINO model
+# The argument `input_sample` is not required in the following cases:
+# you have run `trainer.fit` before trace
+# The Model has `example_input_array` set
+ov_model = Trainer.trace(model, accelerator='openvino', input_sample=torch.randn((1, 3, 256, 256)))
+
+# The usage is almost the same with any PyTorch module
+y_hat = ov_model(torch.rand((10, 3, 256, 256)))
```
Comment (zhentaocc) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_openvino.md:63]: use real image to run inference instead of dummy input. Maybe `image = train_data[0]`?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_inc.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+])
+train_data = datasets.MNIST(
+    root=".",
+    train=True,
+    transform=data_transform
+)
+data_loader = DataLoader(train_data)
+# (Optional) Something else, like training ...
+trainer = Trainer()
+trainer.fit(model, data_loader)
```
Comment (zhentaocc) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_inc.md:50]: Same here.

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_inc.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+### **Step 2: Quantization using Intel Neural Compressor**
+Quantization is widely used to compress models to a lower precision, which not only reduces the model size but also accelerates inference. BigDL-Nano provides `Trainer.quantize()` API for users to quickly obtain a quantized model with accuracy control by specifying a few arguments.
+
+Without extra accelerator, `Trainer.quantize()` returns a pytorch module with desired precision and accuracy. You can add quantization as below:
+```python
+q_model = trainer.quantize(model, calib_dataloader=data_loader)
+# run simple prediction with transparent acceleration
+y_hat = q_model(torch.rand((10, 3, 224, 224)))
```
Comment (zhentaocc) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_inc.md:60]: same here

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_train_quickstart.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    )
+    return cifar10_dm
+
+datamodule = prepare_data(".", 32)
+```
+
+### Step 4: Fit with Nano PyTorch Trainer
+```python
+model = LitResnet(learning_rate=0.05)
+single_trainer = Trainer(max_epochs=1)
```
Comment (zhentaocc) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_train_quickstart.md:101]: Shall we have large `max_epochs`? @yangw1234

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_inc.md (hunk 3 above)
Comment (y199387) [reply docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_inc.md:50]: Do you mean that we don't need fine-tuning before quantization in this doc?

Comment (zhentaocc) [reply docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_inc.md:50]: As it is a tutorial, maybe we still need to make sure the model is well trained. So I suggest using a pre-trained model on MNIST(if you can find one) directly instead of calling 'fit' to do any training-related work.

 Otherwise, train a small model. But is Mobilenet suitable here for training on MNIST and training for only on epoch? If using MNIST, maybe refer to https://github.com/pytorch/examples/blob/master/mnist/main.py ?

Do we need to show the accuracy diff? @yangw1234

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_openvino.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 # BigDL-Nano PyTorch OpenVINO Acceleration Quickstart
 
-**In this guide we will describe how to apply OpenVINO Acceleration on inference pipeline with the APIs delivered by BigDL-Nano in 2 simple steps**
+**In this guide we will describe how to apply OpenVINO Acceleration on inference pipeline with the APIs delivered by BigDL-Nano in 3 simple steps**
```
Comment (qiuxin2012) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_openvino.md:3]: 0 to 3 is 4 steps?

Comment (y199387) [reply docs/readthedocs/source/doc/Nano/QuickStart/pytorch_openvino.md:3]: I saw the Quickstart documentation for Chronos is calculated this way.
https://bigdl.readthedocs.io/en/latest/doc/Chronos/QuickStart/chronos-tsdataset-forecaster-quickstart.html

**Comment by y199387** [comment]
Test docs on website
https://mingtestdocs.readthedocs.io/en/latest/index.html

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_train_quickstart.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+single_trainer.fit(model, datamodule=datamodule)
+```
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_train_quickstart.md:103]: Maybe we can add here
"At this stage, you may already experience some speedup due to the optimized environment variables set by `source bigdl-nano-init`"

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_train_quickstart.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+Besides, you can enable optimizations delivered by BigDL-Nano by setting a paramter or calling a method to accelerate PyTorch or PyTorch Lightning application on training workloads.
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_train_quickstart.md:104]: "You can also enable other optimizations ..."

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+```bash
+conda create py37 python==3.7.10 setuptools==58.0.4
+conda activate py37
+# nightly bulit version
+pip install --pre --upgrade bigdl-nano[pytorch]
+# set env variables for your conda environment
+source bigdl-nano-init
+```
+
+Before you start with onnxruntime accelerator, you are required to install some onnx packages as follows to set up your environment with ONNXRuntime acceleration.
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md:17]: "you are required" -> "you need to"

"you are required" sounds a little impolite to me, what do you think?

Comment (y199387) [reply docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md:17]: Yes, I will change the expression in next update.

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+```bash
+pip install onnx onnxruntime
+```
+### **Step 1: Load the data**
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md:21]: "Here we use two images from the dog_vs_cat dataset from kaggle. You can get the images from here (link)"

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+```python
+from torchvision.io import read_image
+from bigdl.nano.pytorch.vision import transforms
+
+paths = ["../Image/cat.jpg", "../Image/dog.jpg"]
+data_transform =  transforms.Compose([transforms.RandomCrop(224),
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md:27]: You should not use data augmentation in the inference pipeline.
Maybe you should use the "weights.transforms()" here (https://pytorch.org/vision/stable/models.html#using-the-pre-trained-models)

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+![](../Image/cat.jpg)
+![](../Image/dog.jpg)
+
+### **Step 2: Prepare the Model**
+```python
+import torch
+import torch.nn as nn
+from torchvision.models import resnet18
+# define your own model
+class Predictor(nn.Module):
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md:46]: Is this `Predictor` class necessary? How about just use `resnet18(pretrained)`

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+# trainer.fit(model, data_loader)
+```
+
+### **Step 3: Apply ONNXRumtime Acceleration**
+When you're ready, you can simply append the following part to enable your ONNXRuntime acceleration.
+```python
+batch = torch.stack([cat, dog])
+predictor = Predictor()
+predictor.eval()
+predictor(batch)
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md:65]: show the prediction result here?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+# trace your model as an ONNXRuntime model
+# The argument `input_sample` is not required in the following cases:
+# you have run `trainer.fit` before trace
+# The Model has `example_input_array` set
+from bigdl.nano.pytorch import Trainer
+ort_predictor = Trainer.trace(predictor, accelerator="onnxruntime", input_sample=batch)
+
+# The usage is almost the same with any PyTorch module
+ort_predictor(batch)
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md:74]: show the prediction result here?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_openvino.md [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+data_transforms = transforms.Compose([
+        transforms.Resize(256),
+        transforms.RandomCrop(224),
+        transforms.RandomHorizontalFlip(),
+        transforms.ColorJitter(brightness=.5, hue=.3),
+        transforms.ToTensor(),
+        transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
+    ])
+data_set = OxfordIIITPet(root="./data/", transform=data_transforms, 
+                         target_transform=transforms.Lambda(lambda label: torch.tensor(label, dtype=torch.long)))
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_openvino.md:37]: add some comments here

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_openvino.md [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+import torch.nn as nn
+# define your own model
+model_ft = resnet18(pretrained=True)
+num_ftrs = model_ft.fc.in_features
+model_ft.fc = nn.Linear(num_ftrs, len(data_set.classes))
+loss_ft = nn.CrossEntropyLoss()
+optimizer_ft = torch.optim.SGD(model_ft.parameters(), lr=0.01, momentum=0.9, weight_decay=5e-4)
+from bigdl.nano.pytorch import Trainer
+model = Trainer.compile(model_ft, loss_ft, optimizer_ft)
+# (Optional) Something else, like training ...
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_openvino.md:54]: let's remove this comment

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_inc_onnx.md [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+# run simple prediction
+batch = torch.stack([data_set[0][0], data_set[1][0]])
+ort_q_model(batch)
+```
+
+Using accelerator='onnxruntime' actually equals to converting the model from Pytorch to ONNX firstly and then do quantization on the converted ONNX model:
+```python
+ort_model = Trainer.trace(model, accelerator='onnruntime', input_sample=x):
+ort_q_model = trainer.quanize(ort_model, accelerator='onnxruntime', calib_dataloader=dataloader)
+```
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_inc_onnx.md:75]: add prediction for quantized model?

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_inc_onnx.md [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+        transforms.ToTensor(),
+        transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
+    ])
+data_set = OxfordIIITPet(root="./data/", transform=data_transforms)
+data_loader = DataLoader(data_set, batch_size=32, shuffle=True)
+```
+
+### **Step 2: Prepare your Model**
+```python
+from torchvision.models import mobilenet_v3_small
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_quantization_inc_onnx.md:42]: change to resnet18

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_openvino.md [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+```bash
+pip install openvino-dev
+```
+
+### **Step 1: Load the data**
+```python
+from torchvision.io import read_image
+from bigdl.nano.pytorch.vision import transforms
+
+paths = ["../Image/cat.jpg", "../Image/dog.jpg"]
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_openvino.md:27]: let's use `OxfordIIITPet`

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+Before you start with onnxruntime accelerator, you are required to install some onnx packages as follows to set up your environment with ONNXRuntime acceleration.
+```bash
+pip install onnx onnxruntime
+```
+### **Step 1: Load the data**
+```python
+from torchvision.io import read_image
+from bigdl.nano.pytorch.vision import transforms
+
+paths = ["../Image/cat.jpg", "../Image/dog.jpg"]
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md:26]: let's use OxfordIIITPet

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+class Predictor(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.resnet18=resnet18(pretrained=True)
+
+    def forward(self, x):
+        y_hat = self.resnet18(x)
+        return y_hat.argmax(dim=1)
+# (Optional) Something else, like training ...
+# trainer = Trainer()
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_onnxruntime.md:55]: let's add training back to keep consistent with quantization tutorials

File: docs/readthedocs/source/doc/Nano/QuickStart/pytorch_openvino.md [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+class Predictor(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.resnet18=resnet18(pretrained=True)
+
+    def forward(self, x):
+        y_hat = self.resnet18(x)
+        return y_hat.argmax(dim=1)
+# (Optional) Something else, like training ...
+# trainer = Trainer()
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Nano/QuickStart/pytorch_openvino.md:56]: let's add training back to keep consistent with quantization tutorials

File: docs/readthedocs/source/doc/Nano/QuickStart/index.md [hunk 23]
```diff
@@ -45,9 +45,9 @@
...
     In this guide we will describe how to obtain a quantized model with the APIs delivered by BigDL-Nano
 
-[Nano_pytorch_training]: <../../../../../../../../python/nano/notebooks/pytorch/cifar10/nano-trainer-example.ipynb>
-[Nano_pytorch_onnxruntime]: <../../../../../../../../python/nano/notebooks/pytorch/tutorial/pytorch_inference_onnx.ipynb>
-[Nano_pytorch_openvino]: <../../../../../../../../python/nano/notebooks/pytorch/tutorial/pytorch_inference_openvino.ipynb>
-[Nano_pytorch_Quantization_inc]: <../../../../../../../../python/nano/notebooks/pytorch/tutorial/pytorch_quantization_inc.ipynb>
-[Nano_pytorch_quantization_inc_onnx]: <../../../../../../../../python/nano/notebooks/pytorch/tutorial/pytorch_quantization_inc.ipynb>
-[Nano_pytorch_quantization_openvino]: <../../../../../../../../python/nano/notebooks/pytorch/tutorial/pytorch_quantization_openvino.ipynb>
+[Nano_pytorch_training]: <https://github.com/intel-analytics/BigDL/blob/main/python/nano/notebooks/pytorch/cifar10/nano-trainer-example.ipynb>
+[Nano_pytorch_onnxruntime]: <https://github.com/y199387/BigDL/tree/add-nano-quick-start-doc/python/nano/notebooks/pytorch/tutorial/pytorch_inference_onnx.ipynb>
```
[the remark this answers is not in the document]
Comment (y199387) [reply docs/readthedocs/source/doc/Nano/QuickStart/index.md:49]: The readdocs website didn't upload the notebooks in this pr as expected,  so I use personal repository to see how the code change shows. They will be changed to BigDL repository when this pr is open.

**Comment by yangw1234** [comment]
we also need to add tests to these notebooks

**Comment by yangw1234** [comment]
LGTM
